#!/usr/bin/env python

PACKAGE = "blob_tracking"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

color_enum = gen.enum(
    [gen.const("HSV",      int_t, 0, "HSV"),
        gen.const("LAB",     int_t, 1, "LAB")], "Color configuration")


gen.add("color", int_t, 0, "Color Configuration", 0, 0, 0, edit_method=color_enum)

gen.add("h_min", int_t, 0, "Hue Value", 0, 0, 179)
gen.add("h_max", int_t, 0, "Hue Value", 179, 0, 179)
gen.add("s_min", int_t, 0, "Saturation Value", 0, 0, 255)
gen.add("s_max", int_t, 0, "Saturation Value", 255, 0, 255)
gen.add("v_min", int_t, 0, "Brightness Value", 0, 0, 255)
gen.add("v_max", int_t, 0, "Brightness Value", 255, 0, 255)

gen.add("blur", int_t, 0, "Blur", 0, 0, 10)
gen.add("sigma", double_t, 0, "Sigma", 2.7, 0, 10)
gen.add("kernel", int_t, 0, "Kernel", 5, 0, 15)

gen.add("show", bool_t, 0, "Show Images", False)


color_selector_enum = gen.enum(
    [gen.const("None", int_t, 0, "None"),
     gen.const("Blue", int_t, 1, "Blue"),
     gen.const("Red", int_t, 2, "Red"),
     gen.const("Green", int_t, 3, "Green"),
     gen.const("Yellow", int_t, 4, "Yellow"),
     gen.const("Orange", int_t, 5, "Orange")], "Color Selector")

gen.add("color_s", int_t, 0, "Color Selector", 0, 0, 5, edit_method=color_selector_enum)


exit(gen.generate(PACKAGE, "blob_tracking", "Limits"))